### PR TITLE
Plotly Integration - acf

### PR DESCRIPTION
**Pull request by mail4umar**
- added acf plot 
- added unit tests

### CODE REVIEW

File: verticapy/plotting/_plotly/acf.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        fig.add_scatter(
+            x=X,
+            y=-Z,
+            mode="lines",
+            marker_color="red",
+            fill="tonexty",
+            hovertemplate="Confidence <br><b>lag</b>: %{x}<br><b>value</b>: %{customdata:.3f}<br><extra></extra>",
+            customdata=np.abs(-Z),
+        )
+        if self.layout["kind"] == "line":
```
Comment (oualib) [inline verticapy/plotting/_plotly/acf.py:102]: is it possible to put anything related to styling in the init_style fun?

File: verticapy/tests/vDataFrame/test_vDF_plot_plotly.py [hunk 2]
```diff
@@ -1287,3 +1302,133 @@ def test_additional_options_custom_height(self, load_plotly, titanic_vd):
...
+            by=["state"],
+            unit="month",
+            method="spearman",
+        )
+        # Assert - checking if correct object created
+        assert (
+            result.layout["xaxis"]["title"]["text"] == test_title
+        ), "X axis label incorrect"
+
+    def test_properties_scatter_points_and_confodence(self, load_plotly, amazon_vd):
```
Comment (oualib) [inline verticapy/tests/vDataFrame/test_vDF_plot_plotly.py:1339]: typo mistake.

File: verticapy/tests/vDataFrame/test_vDF_plot_plotly.py [hunk 3]
```diff
@@ -1287,3 +1302,133 @@ def test_additional_options_custom_height(self, load_plotly, titanic_vd):
...
+            method="spearman",
+        )
+        # Assert - checking if correct object created
+        assert type(result) == plotly.graph_objs._figure.Figure, "wrong object crated"
+
+    def test_properties_xaxis_label(self, load_plotly, amazon_vd):
+        # Arrange
+        test_title = "Lag"
+        # Act
+        result = amazon_vd.acf(
```
Comment (oualib) [inline verticapy/tests/vDataFrame/test_vDF_plot_plotly.py:1326]: you're computing act in each of the sub function making it very expensive. Why not computing it one time and sharing it between the functions?

**Review by oualib — CHANGES_REQUESTED** [review]
Please try to correct the different points I've mentioned.

File: verticapy/plotting/_plotly/acf.py (hunk 1 above)
Comment (mail4umar) [reply verticapy/plotting/_plotly/acf.py:108]: did it as much as possible.

Comment (mail4umar) [reply verticapy/plotting/_plotly/acf.py:108]: after your comment

File: verticapy/tests/vDataFrame/test_vDF_plot_plotly.py (hunk 3 above)
Comment (mail4umar) [reply verticapy/tests/vDataFrame/test_vDF_plot_plotly.py]: optimized as much as possible for ACF

**Review by oualib — APPROVED** [review]

**Pull request merged**